doc/source/whatsnew/v2.0.0.rst

```diff
@@ -49,6 +49,7 @@ Other enhancements
 - Fix ``test`` optional_extra by adding missing test package ``pytest-asyncio`` (:issue:`48361`)
 - :func:`DataFrame.astype` exception message thrown improved to include column name when type conversion is not possible. (:issue:`47571`)
 - :meth:`DataFrame.to_json` now supports a ``mode`` keyword with supported inputs 'w' and 'a'. Defaulting to 'w', 'a' can be used when lines=True and orient='records' to append record oriented json lines to an existing json file. (:issue:`35849`)
+- :func:`to_datetime` now skips ``datetime.datetime`` and :class:`Timestamp` objects instead of raising a ``ValueError``. (:issue:`49298`)
```

Comment:
let's note that it does this when the user passes `'format'` (I think it already did it in other cases?)

Reply:
You're right, I'll work on it and include your example in a test